src/tests/integration-test

```diff
@@ -1610,6 +1610,112 @@ class Polkit(UDisksTestCase, test_polkitd.PolkitTestCase):
 
 # ----------------------------------------------------------------------------
 
+
+def setup_lio():
+    """create virtual test devices"""
+
+    orig_devs = {dev for dev in os.listdir("/dev") if
+                 re.match(r'sd[a-z]+$', dev)}
+
+    # create fake SCSI hard drives
+    assert subprocess.call(
+        ["targetcli",
+         "restoreconfig dbus-tests/targetcli_config.json"]) == 0
+    time.sleep(0.1)
+
+    devs = {dev for dev in os.listdir("/dev") if
+            re.match(r'sd[a-z]+$', dev)}
+
+    vdevs = ["/dev/%s" % dev for dev in (devs - orig_devs)]
+
+    # let's be 100% sure that we pick a virtual one
+    for d in vdevs:
+        with open('/sys/block/%s/device/model' %
+                  os.path.basename(d)) as model_file:
+            assert model_file.read().strip() == 'udisks_test_dis'
+
+    assert len(vdevs) >= 4
+    return vdevs
+
+
+def remove_lio():
+    # remove the fake SCSI devices and their backing files
+    assert subprocess.call(['targetcli', 'clearconfig confirm=True']) == 0
+    for disk_file in glob("/var/tmp/udisks_test_disk*"):
+        os.unlink(disk_file)
+
+
+class MDRaid(UDisksTestCase):
+    """
+    Test some of the code we are refactoring.
+    """
+    devices = None
+
+    def setUp(self):
+        self.assertTrue(len(glob('/dev/md/int_test*')) == 0)
+        self.md_device = '/dev/md/int_test0'
+
+        self.md_legs = self.devices[:2]
+        create = ['mdadm', '--create', self.md_device, '--level=1',
+                  '--metadata=0.90', '--raid-devices=2']
+        create.extend(self.md_legs)
+        self.assertTrue(subprocess.call(create) == 0)
+        self.sync()
+
+    def tearDown(self):
+        self.assertTrue(
+            subprocess.call(['mdadm', '--stop', self.md_device]) == 0)
+        zero = ['mdadm', '--zero-superblock', '--force']
+        zero.extend(self.md_legs)
+        self.assertTrue(subprocess.call(zero) == 0)
+
+    def test_md_raid_methods(self):
+        self.assertTrue(self.devices is not None)
+        found_one = False
+
+        object_manager = self.client.get_object_manager()
+        objects = object_manager.get_objects()
+        for o in objects:
```

Comment:
If you're advocating that we shouldn't use the dbus API because it's more difficult that using lsblk or mdadm then how do we expect our users to use the dbus API?